src/buildblock/interpolate_projdata.cxx

```diff
@@ -170,7 +174,8 @@ using namespace detail_interpolate_projdata;
   
 Succeeded 
 interpolate_projdata(ProjData& proj_data_out,
-                     const ProjData& proj_data_in, const BSpline::BSplineType these_types,
+                     const shared_ptr<ProjData> proj_data_in,
```

Comment:
why did you change all these types to `shared_ptr`? It is recommended C++ practice to pass references unless you really need the pointer (e.g. because the function needs to store it after the function exists). Reasons include:
- clearer interface: the user knows you don't need access afterwards
- less dependency on `shared_ptr` (I could have a `unique_ptr` for instance, which I wouldn't be able to pass).

Of course, STIR isn't quite consistent in this choice, and it is confusing that we mix conventions. But I see no good reason to change the calling conventions here (certainly as `proj_data_out` is still a reference), but I might be missing something.

If we do need to change it, it should be a `const shared_ptr<const ProjData>`.